pkg/collector/reconcile/horizontalpodautoscaler.go

```diff
@@ -83,64 +83,27 @@ func expectedHorizontalPodAutoscalers(ctx context.Context, params Params, expect
 			return fmt.Errorf("failed to get %w", err)
 		}
 
-		updated := existing.DeepCopyObject().(client.Object)
-		updated.SetOwnerReferences(desired.GetOwnerReferences())
-		setAutoscalerSpec(params, autoscalingVersion, updated)
-
-		annotations := updated.GetAnnotations()
```

Comment:
This removal seems suspicious (handling of annotations and owner references). Could we keep this code and change the copying of the spec instead?

Reply:
Thanks for the review! Hmm it's not clear to me why these need to be manually copied.  Shouldn't these be taken care of  when the desired object is created in this [function](https://github.com/open-telemetry/opentelemetry-operator/blob/main/pkg/collector/horizontalpodautoscaler.go#L36)?  Maybe I'm just confusing myself. As another example it isn't clear to me why `setAutoscalerSpec` is necessary, when these fields are set when the HPA is created. Is there a case I'm not understanding where these fields could be overwritten or missing?

Reply:
I am using your suggestion for now to just do the correct update in setAutoscalerSpec , but am certainly interested in knowing whether there should be a followup PR to clean up the way this update is completed (maybe I'm overthinking this).

Reply:
Hey @pavolloffay, I think I better understand why the old annotations and ownerReferences should be copied instead of relying on the new ones generated when the desired HPA is created. 

I updated the PR once again with a simplified version of copying the spec. Now I am simply assigning the desired MetricSpec to the updated MetricSpec. Can you please take a look? Thanks!